src/parser/types.rs

```diff
@@ -486,6 +503,7 @@ impl Default for FuncOrMethod {
 pub struct FnSig {
     pub kind: FuncOrMethod,
     pub name: Ident,
+    pub generics: HashSet<String>,
```

Comment:
should be `HashSet<Ident>` imo

Reply:
I mentioned this in another PR about the potential issue of using Ident. In some cases, same parameters can exist at different span.

Reply:
BTW it's fine to leave it as is for now, but I think it might also be fine to use the first Ident we see for this type parameter, and then implement Hash manually on the type